packages/lesswrong/components/comments/RecentComments.tsx

```diff
@@ -23,16 +23,19 @@ const RecentComments = ({classes, terms, truncated=false, showPinnedOnProfile=fa
     fragmentName: 'CommentsListWithParentMetadata',
     enableTotal: false,
   });
-  if (!loadingInitial && results && !results.length) {
+  // Filter out comments where the user doesn't have access to the post or tag (mostly for posts that are converted to draft)
+  const validResults = results?.filter(comment => comment.post?._id || comment.tag?._id)
```

Comment:
It would be nicer to filter this out in the view, but joining with the post makes this fiddly and this is enough of an edge case that I decided it wasn't worth it. Also I think there is no real permissions issue here, it seems fine to be able to access comments that were on a post that has been converted to draft if you really want to (you still can't access the post)